presto-docs/src/main/sphinx/develop/client-request-filter.rst

```diff
@@ -0,0 +1,23 @@
+
+======================
+Client Request Filter
+======================
+
+Presto allows operators to customize the headers used by the Presto runtime to process queries. Some example use cases include customized authentication workflows, or enriching query attributes such as the query source. This can be achieved using the Client Request Filter plugin, which allows control over header customization during query execution.
+
+Implementation
+--------------
+
+The ``ClientRequestFilterFactory`` is responsible for creating instances of ``ClientRequestFilter``. It also defines 
+the name of the filter.
+
+``ClientRequestFilter``  interface provides two methods: ``getExtraHeaders()``,
+which returns header values as key-value pairs in a map, and ``getHeaderNames()``, which returns a list of header names used as the header names in client requests.
```

Comment:
I imagine `getExtraHeaders` is used for the runtime to quickly check if it needs to apply a more expensive call to enrich the headers.  If so, add this context to the documentation.

Reply:
Added.